### DISCUSSION TITLE
Wrong skipping operator with NONE_FAILED_MIN_ONE_SUCCESS trigger rule after ShortCircuitOperator

### DISCUSSION THREAD

**Discussion by morooshka** [apache/airflow#28501] [Q&A]
### Apache Airflow version

2.5.0

### What happened

Operator with NONE_FAILED_MIN_ONE_SUCCESS trigger rule being wrongly skipped after the first parent operator had been skipped with its parent ShortCircuitOperator but some other parent operators are still being working. The same is reproducing with ONE_SUCCESS trigger rule

### What you think should happen instead

The operator with NONE_FAILED_MIN_ONE_SUCCESS (ONE_SUCCESS) trigger rule should wait for all parents (one success parent) operators

### How to reproduce

Please find the DAG example:

```
from datetime import datetime
from time import sleep
from airflow import DAG
from airflow.utils.trigger_rule import TriggerRule
from airflow.operators.dummy import DummyOperator
from airflow.operators.python import PythonOperator
from airflow.operators.python import ShortCircuitOperator


def do_sleep() -> bool:
    sleep(1)
    return True


with DAG(
    dag_id="wrong_skip",
    description='Wrong skip',
    schedule_interval=None,
    start_date=datetime(2022, 11, 29),
    default_args={},
    max_active_runs=1,
    catchup=False,
    params={},
    is_paused_upon_creation=True,
    tags=['sandbox']
) as dag:
    task_switcher = ShortCircuitOperator(task_id="switcher", python_callable=lambda: False)
    task_skipped = PythonOperator(task_id="skipped", python_callable=lambda: True)
    task_executed = PythonOperator(task_id="executed", python_callable=do_sleep)
    task_end = DummyOperator(task_id='problem', trigger_rule=TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS)

    task_switcher >> task_skipped >> task_end
    task_executed >> task_end

```

The result of execution:
![image](https://user-images.githubusercontent.com/72786791/208738725-8c8f1b58-7020-4938-a272-c53a33d5efb4.png)

### Operating System

CentOS Linux 7 (Core)

### Deployment

Virtualenv installation

### Anything else

The problem reproduces every time

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by eladkal** [comment]
It's not a bug. It's by desgin.
Please read ShortCircuitOperator documntation.
You will find explnation and how to get the functionality you want :)

https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/python/index.html#airflow.operators.python.ShortCircuitOperator

**Comment by morooshka** [comment]
@eladkal thank you for such a quick answer! It figures out that it was a bug in `apache-airflow==2.2.4` which class definition I was reading  and later a `ignore_downstream_trigger_rules` param was introduced to fix this behaviour

**Reply by eladkal** [reply]
Just to clarify this was never a bug. The operator was designed to be aggressive and ignore downstream trigger rules, later we found that we can make this behavior optional thus a control parameter was added.
If intresting to you then you can read about the history of it in https://github.com/apache/airflow/issues/7858#issuecomment-615863490

**Reply by morooshka** [reply]
@eladkal thank you for your message. I was guiding this class description below (it is from 2.2.4, not from 2.5.0 as in the issue - this is my fault) and it leaded me to misunderstanding. I thought it is a handy version of the BranchPythonOperator
Great that now its clear. Thank you for your help!

```
class ShortCircuitOperator(PythonOperator, SkipMixin):
    """
    Allows a workflow to continue only if a condition is met. Otherwise, the
    workflow "short-circuits" and downstream tasks are skipped.

    The ShortCircuitOperator is derived from the PythonOperator. It evaluates a
    condition and short-circuits the workflow if the condition is False. Any
    downstream tasks are marked with a state of "skipped". If the condition is
    True, downstream tasks proceed as normal.

    The condition is determined by the result of `python_callable`.
    """
```
